api/api.js

```diff
@@ -1534,6 +1562,68 @@ export default (router) => {
     }
   })
 
+  router.get('/hubs/:publicKeyOrHandle/releases/archived', async (ctx) => {
+    try {
+      await NinaProcessor.init()
+      let { offset=0, limit=BIG_LIMIT, sort='desc', column='datetime' } = ctx.query;
+      column = formatColumnForJsonFields(column);
+      const hub = await hubForPublicKeyOrHandle(ctx)
+      let releases
+      const archivedReleasesForHub = await Release
+        .query()
+        .joinRelated('hubs')
+        .where('hubs_join.hubId', hub.id)
+        .where('hubs_join.visible', false)
+        .where(ref('metadata:name').castText(), 'ilike', `%${query}%`)
+        .orderBy(column, sort)
+        .range(Number(offset), Number(offset) + Number(limit) - 1);
+      releases = await Release.query()
+        .whereIn('id', archivedReleasesForHub.results.map(release => release.id))
+
+      releases = {
```

Comment:
Instead of redefining it again, you can just combine it to something like:

```
const releases = {
  results: await Release.query()
    .whereIn('id', archivedReleasesForHub.results.map(release => release.id)),
  total: archivedReleasesForHub.total
};```

Reply:
should be cleaned up now - turns out the second queries werent even necessary